src/app-data/index.ts

```diff
@@ -1,5 +1,30 @@
 import type { BuildConditionals } from '@stencil/core/internal';
 
+/**
+ * A collection of default build flags for a Stencil project.
+ *
+ * This collection can be found throughout the Stencil codebase, often imported from the `@app-data` module like so:
+ * ```ts
+ * import { BUILD } from '@app-data';
+ * ```
+ * and is used to determine if a portion of the output of a Stencil _project_'s compilation step can be eliminated.
+ *
+ * e.g. When `BUILD.allRenderFn` evaluates to `false`, the compiler will eliminate conditional statements like:
+ * ```ts
+ * if (BUILD.allRenderFn) {
+ *   // some code that will be eliminated if BUILD.allRenderFn is false
+ * }
+ * ```
+ *
+ * `@app-data`, the module that `BUILD` is imported from, is an alias for the `@stencil/core/internal/app-data`, and is
+ * partially referenced by {@link STENCIL_APP_DATA_ID}. The `src/compiler/bundle/app-data-plugin.ts` references
+ * `STENCIL_APP_DATA_ID` uses it to replace these defaults with {@link BuildConditionals} that are derived from a
+ * Stencil project's contents (i.e. metadata from the components). This replacement happens at a Stencil project's
+ * compile time. Such code can be found at `src/compiler/app-core/app-data.ts`.
+ *
+ *
+ * This collection is expected to be updated before used in compilation of a Stencil project.
+ */
```

Comment:
```suggestion
 *
 * This collection should be updated before use in compilation of a Stencil project.
 */
```

I'm not sure what the last sentence there means exactly, does that change sort of get at it? I'm not clear maybe who is supposed to update it, so it's not clear to me what this refers to

Reply:
I had this sentence in as an early placeholder before I added [this whole bit](https://github.com/ionic-team/stencil/pull/4624/files#diff-d01a21edbe6c4684280903b1fc7ef9d625c04ab58e14f0a0ae976276fabf1a63R6-R23). It think it probably makes sense to drop it at this point - 047bdb2f0